instill/model/v1alpha/model.proto

```diff
@@ -0,0 +1,403 @@
+  }
+
+  // Model version
+  uint64 version = 1;
+  // Model version corresponding model ID
+  uint64 model_id = 2;
+  // Model version description
+  string description = 3;
+  // Model version creation time
+  google.protobuf.Timestamp created_at = 4;
+  // Model version update time
+  google.protobuf.Timestamp updated_at = 5;
+  // Model version status
+  Status status = 6;
+}
+
+// Model represents the content of a model
+message Model {
+
+  // Task enumerates the task type of a model
+  enum Task {
+    // Task: UNSPECIFIED
+    TASK_UNSPECIFIED = 0;
+    // Task: CLASSIFICATION
+    TASK_CLASSIFICATION = 1;
+    // Task: DETECTION
+    TASK_DETECTION = 2;
+  }
+
+  // Model ID
+  uint64 id = 1 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model name
+  string name = 2;
+  // Model full name (i.e., [user_name/org_name]/model_name)
+  string full_name = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model task
+  Task task = 4;
+  // Model versions
+  repeated ModelVersion model_versions = 5;
+}
+
+// ModelInitData represents the required data for initialising a model instance
+message ModelInitData {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // Model content in byte
+  bytes byte = 2 [ (google.api.field_behavior) = REQUIRED ];
+  // Model description
+  string description = 3 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model task type
+  Model.Task task = 4 [ (google.api.field_behavior) = OPTIONAL ];
+}
+
+// CreateModelRequest represents a request to create a model
+message CreateModelRequest {
+  // Model initialising data
+  ModelInitData model_init_data = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelResponse represents a response for a model instance
+message CreateModelResponse {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelBinaryFileUploadRequest represents a request to create a model
+message CreateModelBinaryFileUploadRequest {
+  // Model initialising data
+  ModelInitData model_init_data = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelBinaryFileUploadResponse represents a response for a model
+// instance
+message CreateModelBinaryFileUploadResponse {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelRequest represents a request to update a model
+message UpdateModelVersionRequest {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // Model version
+  uint64 version = 2 [ (google.api.field_behavior) = REQUIRED ];
+  // Model version description
+  string description = 3;
+  // Model version status
+  ModelVersion.Status status = 4;
+  // Update mask for a model instance
+  google.protobuf.FieldMask field_mask = 5
+      [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// UpdateModelVersionResponse represents a response for a model instance
+message UpdateModelVersionResponse {
```

Comment:
@pinglin @xiaofei-du I think it should return ModelVersion instead of model. What do you think?

Reply:
Yeah. That makes sense.